packages/plugin-core/src/test/suite-integ/migration.test.ts

```diff
@@ -338,64 +342,163 @@ suite("Migration", function () {
       });
     });
 
-    test("migrate to 0.70.0 (preview namespace migration)", (done) => {
-      DendronExtension.version = () => "0.70.0";
-      runLegacyMultiWorkspaceTest({
+    describeMultiWS(
```

Comment:
The past three migrations (for each namespaces) were replaced in-place, since on subsequent migrations we were removing the type definitions of the config namespace prior. We didn't account for backwards compatibility and keeping track of each config version would have compounded exponentially where we are retrieving the config values. This will be true for this time as well, but not until we decide we are going to automatically run the migration on activation and completely remove backwards compatibility.

I can revert this so that the v3 -> v4 migration (0.70 migration) is there, and we just add a new migration that does v4 -> v5 on top of that, but this will have to be reverted again and the logic will have to change again.

If you look at the 0.81 migration, you can see that the logic build on top of 0.70 migration and includes everything it does in 0.70. if the user upgrades to a version prior to the introduction of this migration, it won't matter. If they upgrade to a version after this migration, the old config migrations won't be omitted since the latest one does everything for them.

Reply:
the older migrations also had problems where the migration code would break because the config namespace was changed, but this was more pronounced in the last three migrations where keeping history of each of the migration and adding new deltas on top didn't make sense. After all the aggressive migration is done, we can wait until the majority of user base is up to date with the config version, and resume our original migration strategy (keeping history of every migration we do) since each subsequent migrations will (hopefully) be much smaller than these.